source/ui/dataman/AMSpectrumAndPeriodicTableView.h

```diff
@@ -0,0 +1,194 @@
+#ifndef AMSPECTRUMANDPERIODICTABLEVIEW_H
+#define AMSPECTRUMANDPERIODICTABLEVIEW_H
+
+#include <QWidget>
+
+#include "beamline/AMDetector.h"
+
+#include "MPlot/MPlot.h"
+#include "MPlot/MPlotWidget.h"
+#include "MPlot/MPlotSeries.h"
+#include "ui/AMTopFrame.h"
+
+#include <QHBoxLayout>
+#include <QVBoxLayout>
+#include <QGridLayout>
+#include <QPushButton>
+#include <QLabel>
+#include <QDoubleSpinBox>
+#include <QButtonGroup>
+
+#include "util/AMSelectablePeriodicTable.h"
+#include "util/AMNameAndRangeValidator.h"
+#include "util/AMRange.h"
+#include "ui/util/AMSelectableElementView.h"
+#include "ui/util/AMSelectablePeriodicTableView.h"
+#include "dataman/AMScanSetModel.h"
+#include "dataman/AMnDIndex.h"
+#include "dataman/AMAxisInfo.h"
+
+/// A helper class containing common functionality between views with a spectrum and periodic table component. Not to be used as a stand-alone class.
+class AMSpectrumAndPeriodicTableView : public QWidget
+{
+	Q_OBJECT
+
+public:
+	AMSpectrumAndPeriodicTableView(QWidget *parent = 0);
+
+	virtual ~AMSpectrumAndPeriodicTableView();
+
+	/// Energy Settings
+	/// Returns the energy range used for filtering.  If no range has been provided the range is null.
+	const AMRange &energyRange() const { return emissionLineValidator_->range(); }
```

Comment:
This should just be an AMRange and not a `const AMRange &`.  It's different when passing in classes than returning.  Passing in variables with it means your are passing a non-modifiable reference, which is very fast.  Returning one requires that reference being valid and if anything happens to that memory you will get a seg fault crash.  This is why return values will create a copy and return that (i.e.: `AMRange energyRange() const;`).